### ISSUE TITLE
PR Destination Filter bug

### ISSUE THREAD

**Issue by KafkaPalazzo** [KyleLNicholls/parameterized-builds#159]
- _Parameterized Builds for Jenkins_ plugin version - Parameterized Builds for Jenkins version 3.1.2
- _Bitbucket_ version - Atlassian Bitbucket v5.2.2
- _Jenkins_ Version - Jenkins ver. 2.121.3

I encountered the issue after I set 3 builds with the triggers MANUAL, PR OPENED, PR MERGED.
Two of the builds used the same PR Destination Filter while the third build used a different one.
The order of the builds in the builds list affected which builds were triggered and when a PR was created or updated.

See the attached file for all the test cases I ran so far.
[pr_dst_filter_cases.txt](https://github.com/KyleLNicholls/parameterized-builds/files/2478645/pr_dst_filter_cases.txt)

Thnx,
Tom.

**Comment by a-hacker** [comment]
Hi @KafkaPalazzo. Thanks for taking the time to submit this issue and providing good details!

I apologize for the delay on addressing this issue. I started running through your test cases on both the latest version of this plugin and the version you are running (3.1.2). However, I was unable to replicate your issue. I only tested against the first four test cases you supplied and against Bitbucket version 5.4.0 and Jenkins version 2.100. Since everything worked as expected for me, I believe there may be something else working incorrectly. Did you happen to see any errors in the Bitbucket logs when opening a PR? 

When you ran the test cases, did you run each case once or did you run them multiple times with the same result? If you only ran them once, I'd appreciate it if you ran each of your test cases a few more times to see if the triggered and untriggered builds are consistent or if some triggers are being erratically lost.

**Comment by KafkaPalazzo** [comment]
Hey @a-hacker, it's all good, thnx for replying,

I ran another batch of tests, multiple times.
It looks like the builds are inspected by order of insertion. 
If the PR Destination Filter of the first build in the list does not match the PR destination branch, the inspection is halted and the next builds in the list aren't inspected (case 2).
If the PR Destination Filter of the first build in the list does match however, the next builds on the list are inspected and may be triggered (case 4). A false match will halt the inspection again.

Another strange thing I noticed is if I remove one of the builds, recreate it at the bottom of the list and save, when I open the hook again, the PR Destination FIlters of the builds get mixed up (Case A).

Thnx,
Tom.

[pr_dst_filter_cases2.txt](https://github.com/KyleLNicholls/parameterized-builds/files/2499101/pr_dst_filter_cases2.txt)

**Comment by a-hacker** [comment]
Hey @KafkaPalazzo, I have to apologize again. I forgot that Bitbucket doesn't allow you to downgrade an already installed plugin so I had been testing against version 3.2.6 the whole time. I just uninstalled the plugin from my bitbucket test instance, installed 3.1.2 and I was able to replicate your issue. 

I don't know which version fixed you original issue but I'd recommend updating to the latest version (3.2.6) to resolve your issue anyways.

As for the new issue with the PR Destination Filters being mixed up (Case A), I still see that being an issue in the latest version. However, I already did a re-write of that code which has yet to be released. I need to do a few more tests but I believe that Case A is already fixed on master. If that is the case, I'll create a new release once I'm confident there are no issues to fix this bug.
